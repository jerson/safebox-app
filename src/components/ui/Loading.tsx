import React from 'react';
import {
  ActivityIndicator,
  ActivityIndicatorProps,
  StyleSheet,
  View,
  ViewStyle
} from 'react-native';
import Colors from '../../modules/constants/Colors';

const styles = StyleSheet.create({
  marginSmall: {
    padding: 5
  } as ViewStyle,
  marginMedium: {
    padding: 10
  } as ViewStyle,
  marginLarge: {
    padding: 20
  } as ViewStyle
});

export interface LoadingProps extends ActivityIndicatorProps {
  margin?: 'small' | 'medium' | 'large';
}

export interface LoadingState {}

export default class Loading extends React.Component<
  LoadingProps,
  LoadingState
> {
  static defaultProps = {
    color: Colors.primary,
    margin: 'none'
  };

  render() {
    const { color, style, margin, size, ...props } = this.props;
    return (
      <View
        style={[
          margin === 'small' && styles.marginSmall,
          margin === 'medium' && styles.marginMedium,
          margin === 'large' && styles.marginLarge,
          style
        ]}
      >
        <ActivityIndicator color={color} size={size} {...props} />
      </View>
    );
  }
}
