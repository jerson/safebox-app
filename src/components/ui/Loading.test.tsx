import React from 'react';

import * as renderer from 'react-test-renderer';
import Loading from './Loading';

it('renders without crashing', () => {
  const snapshot = renderer.create(<Loading />).toJSON();
  expect(snapshot).toBeTruthy();
});

it('render just views', () => {
  const snapshot = renderer.create(<Loading />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
