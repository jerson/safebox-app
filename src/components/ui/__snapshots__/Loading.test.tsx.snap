// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Array [
      false,
      false,
      false,
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#4531cd"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
